### DISCUSSION TITLE
Use QML forms in Qfield ?

### DISCUSSION THREAD

**Discussion by jfmoyen** [opengisch/QField#2719] [Q&A]
In relation to this issue (https://github.com/opengisch/QField/issues/1882), I'm actually prepared to try and see if I can teach myself enough QML to call QtSensors (that appear to be available in QField) and get a compass reading. I would then use this in an attribute form to populate the relevant attributes.

However, it is unclear to me whether QField actually supports user-defined QML widgets (the ones that are defined in the attribute form of QGIS) ? Quick and dirty tests using the "hello word" widget would appear to suggest otherwise, but I could not find an explicit answer ?

**Comment by suricactus** [comment]
The short answer is "yes", QField :green_heart: QML & HTML forms https://github.com/opengisch/QField/pull/1221 .

**Comment by jfmoyen** [comment]
It does ? Lovely ! My two-minute test suggested otherwise, but obviously I was too quick to give up :-)
